src/dotnet/MsbuildProject.cs

```diff
@@ -62,6 +62,19 @@ public static MsbuildProject FromFile(ProjectCollection projects, string project
 
         public static MsbuildProject FromDirectory(ProjectCollection projects, string projectDirectory)
         {
+            FileInfo projectFile = GetProjectFileFromDirectory(projectDirectory);
+
+            var project = TryOpenProject(projects, projectFile.FullName);
```

Comment:
@krwq - The matter is that TryXXX APIs have a certain expectation associated with them, even if you understand them to be the same. It is known (and expected) by people who see a Try API that it will return a bool and a out param of the type you are trying to get.